pkg/service/credential/storage/bolt.go

```diff
@@ -1,35 +1,138 @@
-	"strings"
-
 	"github.com/TBD54566975/ssi-sdk/credential/signing"
 	"github.com/goccy/go-json"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"math/big"
+	"strings"
 
 	"github.com/tbd54566975/ssi-service/internal/util"
 	"github.com/tbd54566975/ssi-service/pkg/storage"
 )
 
 const (
-	namespace                = "credential"
+	credentialNamespace           = "credential"
+	statusListCredentialNamespace = "status-list-credential"
+	statusListIndexNamespace      = "status-list-index"
+
 	credentialNotFoundErrMsg = "credential not found"
 )
 
 type BoltCredentialStorage struct {
 	db *storage.BoltDB
 }
 
+type StatusListIndex struct {
+	Index int `json:"index"`
+}
+
 func NewBoltCredentialStorage(db *storage.BoltDB) (*BoltCredentialStorage, error) {
 	if db == nil {
 		return nil, errors.New("bolt db reference is nil")
 	}
+
+	// TODO: (Neal) there is a current bug with our Bolt implementation where if we do a GET without anything in the db it will throw an error
+	// Doing initial writes and then deleting will "warm up" our database and when we do a GET after that it will not crash and return empty list
+	err := db.Write(credentialNamespace, "fake-key", nil)
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "problem writing status initial write to db")
+	}
+
+	err = db.Delete(credentialNamespace, "fake-key")
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "problem with initial delete to db")
+	}
+
+	err = db.Write(statusListCredentialNamespace, "fake-key", nil)
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "problem writing status initial write to db")
+	}
+
+	err = db.Delete(statusListCredentialNamespace, "fake-key")
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "problem with initial delete to db")
+	}
+
+	randUniqueList, err := randomUniqueNum(128000)
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "problem with generating random unique list")
+	}
+
+	uniqueNumBytes, err := json.Marshal(randUniqueList)
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "could not marshal random unique numbers")
+	}
+
+	err = db.Write(statusListIndexNamespace, "status-list-indexes", uniqueNumBytes)
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "problem writing status list indexes to db")
+	}
+
+	statusListIndexBytes, err := json.Marshal(StatusListIndex{Index: 0})
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "could not marshal status list index bytes")
+	}
+
+	err = db.Write(statusListIndexNamespace, "current-list-index", statusListIndexBytes)
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "problem writing current list index to db")
+	}
+
 	return &BoltCredentialStorage{db: db}, nil
 }
 
+func (b BoltCredentialStorage) GetNextStatusListRandomIndex() (int, error) {
+
+	gotUniqueNumBytes, err := b.db.Read(statusListIndexNamespace, "status-list-indexes")
+	if err != nil {
+		return -1, util.LoggingErrorMsgf(err, "could not get unique numbers from db")
```

Comment:
I typically recommend having more succinct error messages, that don't include synonyms of error (e.g. "could not", "failed", "break", "fault").

The result is logs that are easier to read, and are not repetitive. Compare the following two log lines:
* `ERROR: could not get unique numbers from db: db not initialized`
* `ERROR: reading status list: db not initialized`.

The second conveys the same meaning, but is easier to read and avoids repetition.

```suggestion
		return -1, util.LoggingErrorMsgf(err, "reading status list")
```

Reply:
need to improve this everywhere